### PR TITLE
CP-2275 Adds support for --fatal-lints analyzer flag

### PR DESCRIPTION
**Pull request by travissanderson-wf**
Addresses https://github.com/Workiva/dart_dev/issues/164

@evanweible-wf @maxwellpeterson-wf @trentgrover-wf @jayudey-wf

### CODE REVIEW

File: lib/src/tasks/analyze/cli.dart [hunk 1]
```diff
@@ -62,9 +66,9 @@ class AnalyzeCli extends TaskCli {
         fatalWarnings: fatalWarnings,
         hints: hints,
         fatalHints: fatalHints,
+        fatalLints: fatalLints,
         strong: strong);
     var title = task.analyzerCommand;
-    if (fatalHints) title += ' (treating hints as fatal)';
```
Comment (travissanderson-wf) [inline lib/src/tasks/analyze/cli.dart]: this was a bit of detritus from when we added our own fatal hints support, figured I would clean it up here

**Comment by evanweible-wf** [comment]
+1 looks great, thanks!

**Comment by travissanderson-wf** [comment]
CI is passing now

**Comment by evanweible-wf** [comment]
+1

**Comment by travissanderson-wf** [comment]
@maxwellpeterson-wf @trentgrover-wf @jayudey-wf would you mind reviewing this?

File: test/integration/analyze_test.dart [hunk 2]
```diff
@@ -130,6 +133,21 @@ void main() {
...
 
+    test('should report lints as fatal if configured to do so', () async {
+      Analysis analysis =
+          await analyzeProject(projectWithLints, fatalLints: true);
+      expect(analysis.numErrors, equals(1));
+      expect(analysis.numHints, equals(0));
+      expect(analysis.exitCode, greaterThan(0));
+    });
+
+    test('should not report hints as fatal if none existed', () async {
```
Comment (jayudey-wf) [inline test/integration/analyze_test.dart]: should this be `should not report lints as fatal if none existed`

as well should you check to make sure that the numErrors are 0

**Comment by travissanderson-wf** [comment]
@jayudey-wf addressed, thanks for the catch on that

**Comment by jayudey-wf** [comment]
smithy failure will be resolved by https://github.com/Workiva/dart_dev/pull/191

**Comment by jayudey-wf** [comment]
+10, newly added tests exercise new change

**Comment by travissanderson-wf** [comment]
travis-ci is green 🏁

**Comment by evanweible-wf** [comment]
+1 @jayudey-wf I think we can go ahead and merge this since the smithy failure is being taken care of separately.

**Comment by jayudey-wf** [comment]
QA Resource Approval: +1

- [ ] Testing instruction (talked with Travis offline)
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - performed and documented by me above
- [ ] Unit test created/updated
- [ ] All unit tests pass 

Merging into master.

**Comment by colefeisthamel-wf** [comment]
RM +1 
updated Rosie functions

**Pull request merged**